File: PowerShellAutomationAgent.API/PowerShellAutomationAgent.API/Services/Jobs/Models/JobResponseModel.cs
using PowerShellAutomationAgent.API.Enums;

namespace PowerShellAutomationAgent.API.Services.Jobs.Models;

public class JobResponseModel
{
    public int Id { get; set; }
    public JobStates State { get; set; }
    public string? Script { get; set; }
    public bool HasArtifacts { get; set; } = false;
    public string ProjectName { get; set; } = string.Empty;
    public DateTimeOffset CreatedOnUtc { get; set; }
    public DateTimeOffset? StartedOnUtc { get; set; }
    public DateTimeOffset? FinishedOnUtc { get; set; }
    public string? Output { get; internal set; }
}

File: PowerShellAutomationAgent.UI/src/services/jobs/models/job-response-model.ts
export interface JobResponseModel {
  id: number;
  state: JobStates;
  script: string | undefined;
  output: string | undefined;
  hasArtifacts: boolean;
  projectName: string;
  createdOnUtc: Date;
  startedOnUtc: Date;
  finishedOnUtc: Date;
}

export enum JobStates {
  Queued = 1,
  InProgress = 2,
  Succeeded = 3,
  Failed = 4,
}